src/map/utils/battleutils.cpp

```diff
@@ -1714,91 +1714,68 @@ namespace battleutils
-            // For mobs, we can assume their skill is capped at their level, so this term is 1 anyway.
-            CCharEntity* PChar = (CCharEntity*)PDefender;
-            float        skill = PChar->GetSkill(PSpell->getSkillType());
-            if (skill <= 0)
-            {
-                skill = 1;
-            }
-
-            float cap = GetMaxSkill((SKILLTYPE)PSpell->getSkillType(), PChar->GetMJob(), PChar->GetMLevel());
+        {
+            CCharEntity* PChar      = (CCharEntity*)PDefender;
+            float        skillCap   = GetMaxSkill((SKILLTYPE)PSpell->getSkillType(), PChar->GetMJob(), PChar->GetMLevel());
+            float        skillLevel = PChar->GetSkill(PSpell->getSkillType());
 
-            // if cap is 0 then player is using a spell from their subjob
-            if (cap == 0)
+            // If skill cap is 0, player may be using a spell from their subjob.
+            if (skillCap == 0)
             {
-                cap = GetMaxSkill((SKILLTYPE)PSpell->getSkillType(), PChar->GetSJob(),
-                                  PChar->GetMLevel()); // This may need to be re-investigated in the future...
+                skillCap = GetMaxSkill((SKILLTYPE)PSpell->getSkillType(), PChar->GetSJob(), PChar->GetMLevel()); // This may need to be re-investigated in the future.
             }
 
-            if (skill > cap)
+            // If skill level is 0, set ratio to 10.
+            if (skillLevel <= 0)
             {
-                skill = cap;
+                skillRatio = 10.0f;
             }
-
-            float ratio = cap / skill;
-            check *= ratio;
-
-            // prevent from spilling over 100 - resulting in players never being interupted
-            if (check > 100)
+            else
             {
-                check = 100;
+                skillRatio = skillCap / skillLevel;
             }
 
-            // apply any merit reduction
+            // Fetch player-only interruption rate reduction from merits.
             meritReduction = ((CCharEntity*)PDefender)->PMeritPoints->GetMeritValue(MERIT_SPELL_INTERUPTION_RATE, (CCharEntity*)PDefender);
         }
 
-        float interruptRate = ((100.0f - (meritReduction + (float)PDefender->getMod(Mod::SPELLINTERRUPT))) / 100.0f);
-
-        float chance = xirand::GetRandomNumber<float>(1.0f);
-
-        // caps, always give a 1% chance of interrupt // TODO: verify, perhaps there is a breakpoint where this no longer happens.
-        if (check < 0.01)
-        {
-            check = 0.01;
-        }
-
         // SIRD reduces the interrupt after all the calculations are done -- as evidenced by the infamous "102% SIRD" builds.
         // Anything less than 102% interrupt results in the ability to be interrupted.
         // Note: the 102% is probably an x/256 x/1024 nonsense -- sometimes 101% works.
-        check *= interruptRate;
+        float SIRDRatio = (101.0f - meritReduction - (float)PDefender->getMod(Mod::SPELLINTERRUPT)) / 100.0f;
```

Comment:
I suppose 101% implements what retail in effect does, but I wonder if we should actually care? we don't for most other things that have similar problems like Haste (25% = 25% and not 26% = 25%)

Reply:
I will trust your judgement here. I placed a 101 independently from what we do in other places preciselly because a 100% does not, in fact, ever give immunity to spell interruption whereas 101/102+ does.

Sowhatever you decide in this matter is ok with me.

Reply:
I think we should be consistent. Their programmers clearly intended 100% = 100%, so we should abide by that. It's not their fault x/256 or x/1024 exists because of PS2 devkits

Reply:
Changed to 100.0f